middleware/cache/cache.go

```diff
@@ -133,6 +149,20 @@ func New(config ...Config) fiber.Handler {
 		e.status = c.Response().StatusCode()
 		e.ctype = utils.CopyBytes(c.Response().Header.ContentType())
 		e.cencoding = utils.CopyBytes(c.Response().Header.Peek(fiber.HeaderContentEncoding))
+		e.e2eHeaders = make(map[string][]byte)
```

Comment:
why convert to string? that would take time

Reply:
As it's impossible to determine both header key and value ahead of time, it feels like a map is needed to compare `ignoreHeaders` with response headers with a O(1) complexity.

If you have another approach to recommend, I take it 🙏